apps/ui-kit/lib/components/mongo-shell/MongoShell.vue

```diff
@@ -0,0 +1,321 @@
+<template>
+  <div class="BksUiKit BksTextEditor BksSqlTextEditor" ref="shell"></div>
+</template>
+
+<script lang="ts">
+import { TextEditor } from '../text-editor/v2/TextEditor';
+import { monokai } from '@uiw/codemirror-theme-monokai';
+import { Annotation, EditorState, Compartment } from '@codemirror/state';
+import { EditorView, keymap } from '@codemirror/view';
+import props from './props';
+import { addOutputEffect, OutputField } from './extensions/ansi-widget';
+import { mongoMode } from './extensions/mongo-mode';
+import { PromptLineField, PromptSymbolField, setPromptLineEffect, setPromptSymbolEffect } from './state';
+import AnsiToHtml from 'ansi-to-html';
+
+const ansiToHtml = new AnsiToHtml();
+
+const SubmitAnnotation = Annotation.define<boolean>();
+const BlockedEdit = Annotation.define();
+
+export default {
+  props,
+  data() {
+    return {
+      shell: null,
+      commandHistory: [],
+      historyIndex: -1,
+      promptLine: 0
+    }
+  },
+  watch: {
+    promptSymbol(value) {
+      if (this.textEditor) {
+        this.textEditor.dispatchChange({
+          effects: [
+            setPromptSymbolEffect.of(value)
+          ]
+        })
+      } else {
+        this.initialize();
+      }
+    },
+    output(value) {
+      const insertPos = this.textEditor.getLength();
+      let output = value.output;
+
+      if (typeof output === 'string' && /\x1b\[[0-9;]*m/.test(output)) {
+        const html = ansiToHtml.toHtml(output);
+        this.textEditor.dispatchChange({
+          changes: { from: insertPos, insert: '\n' },
+          effects: addOutputEffect.of({ pos: insertPos + 1, html }),
+          scrollIntoView: true
+        });
+      } else if (typeof output === 'object') {
+        let formattedOutput;
+        try {
+          formattedOutput = JSON.stringify(output, null, 2);
+        } catch (err) {
+          console.warn('Failed to stringify JSON output for mongo shell: ', err.message);
+          formattedOutput = JSON.stringify(value.output);
+        }
+        this.textEditor.dispatchChange({
+          changes: { from: insertPos, insert: `\n${formattedOutput}`},
+          scrollIntoView: true
+        })
+      } else {
+        this.textEditor.dispatchChange({
+          changes: { from: insertPos, insert: `\n${output}`},
+          scrollIntoView: true
+        })
+      }
+
+      // this will need to update the shell prompt first
+      this.resetPrompt();
+    }
+  },
+  computed: {
+    replaceExtensions() {
```

Comment:
I'm fine with this, but another way to add extensions is by making a new class for Mongo Shell, via `getExtensions()`. See [SQLTextEditor](https://github.com/beekeeper-studio/beekeeper-studio/blob/master/apps/ui-kit/lib/components/sql-text-editor/v2/SqlTextEditor.ts#L50) for example.